File: packages/sdk/sdk-php/tests/workspace_mutations_test.php
<?php
declare(strict_types=1);

require_once __DIR__ . "/../src/index.php";

use AIStats\Sdk\AIStats;

function assert_true(bool $condition, string $message): void
{
    if (!$condition) {
        throw new RuntimeException($message);
    }
}

final class FakeWorkspaceMutationsClient extends \AIStats\Gen\Client
{
    public array $calls = [];

    public function __construct()
    {
        parent::__construct("https://example.test");
    }

    public function request(string $method, string $path, ?array $query = null, ?array $headers = null, $body = null)
    {
        $this->calls[] = [$method, $path, $query, $headers, $body];
        return match ([$method, $path]) {
            ["POST", "/workspaces"] => ["data" => ["id" => "ws_123", "slug" => "sandbox", "name" => "Sandbox Workspace"]],
            ["PATCH", "/workspaces/ws_123"] => ["data" => ["id" => "ws_123", "slug" => "sandbox", "name" => "Renamed Workspace", "archived" => true]],
            ["DELETE", "/workspaces/ws_123"] => ["data" => ["id" => "ws_123", "deleted" => true]],
            default => throw new RuntimeException("Unexpected request"),
        };
    }
}

$client = new AIStats(
    apiKey: "test",
    basePath: "https://api.phaseo.app/v1",
    enableDeprecationWarnings: false
);

$fake = new FakeWorkspaceMutationsClient();
$ref = new ReflectionClass($client);
$prop = $ref->getProperty("client");
$prop->setAccessible(true);
$prop->setValue($client, $fake);

$created = $client->createWorkspace(["name" => "Sandbox Workspace", "slug" => "sandbox"]);
$updated = $client->updateWorkspace("ws_123", ["name" => "Renamed Workspace", "archived" => true]);
$deleted = $client->deleteWorkspace("ws_123");

assert_true(($created["data"]["slug"] ?? null) === "sandbox", "expected created workspace slug");
assert_true(($updated["data"]["archived"] ?? null) === true, "expected updated workspace archived flag");
assert_true(($deleted["data"]["deleted"] ?? null) === true, "expected deleted workspace payload");
assert_true($fake->calls === [
    ["POST", "/workspaces", null, null, ["name" => "Sandbox Workspace", "slug" => "sandbox"]],
    ["PATCH", "/workspaces/ws_123", null, null, ["name" => "Renamed Workspace", "archived" => true]],
    ["DELETE", "/workspaces/ws_123", null, null, null],
], "expected request call sequence for workspace mutations");

echo "php workspace mutation tests ok" . PHP_EOL;

File: packages/sdk/sdk-ruby/tests/workspace_mutations_test.rb
require "minitest/autorun"
require_relative "../lib/index"

class WorkspaceMutationsTest < Minitest::Test
  def test_workspace_mutation_helpers_return_payloads
    client = AIStatsSdk::AIStats.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      case [method, path]
      when ["POST", "/workspaces"]
        { "data" => { "id" => "ws_123", "slug" => "sandbox", "name" => "Sandbox Workspace" } }
      when ["PATCH", "/workspaces/ws_123"]
        { "data" => { "id" => "ws_123", "slug" => "sandbox", "name" => "Renamed Workspace", "archived" => true } }
      when ["DELETE", "/workspaces/ws_123"]
        { "data" => { "id" => "ws_123", "deleted" => true } }
      else
        raise "Unexpected request"
      end
    end

    created = client.create_workspace("name" => "Sandbox Workspace", "slug" => "sandbox")
    updated = client.update_workspace("ws_123", "name" => "Renamed Workspace", "archived" => true)
    deleted = client.delete_workspace("ws_123")

    assert_equal "sandbox", created["data"]["slug"]
    assert_equal true, updated["data"]["archived"]
    assert_equal true, deleted["data"]["deleted"]
    assert_equal [
      ["POST", "/workspaces", nil, nil, { "name" => "Sandbox Workspace", "slug" => "sandbox" }],
      ["PATCH", "/workspaces/ws_123", nil, nil, { "name" => "Renamed Workspace", "archived" => true }],
      ["DELETE", "/workspaces/ws_123", nil, nil, nil]
    ], calls
  end
end
